src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1588,19 +1588,19 @@ public static unsafe int SequenceCompareTo(ref byte first, int firstLength, ref
                     // Invert matches to find differences
                     uint differences = ~matches;
                     // Find bitflag offset of first difference and add to current offset
-                    offset = (IntPtr)((int)(byte*)offset + BitOperations.TrailingZeroCount((int)differences));
+                    offset += (nuint)BitOperations.TrailingZeroCount(differences);
```

Comment:
Recommend using `(uint)BitOperations.TrailingZeroCount(...)` to avoid unnecessary sign-extension. `uint` will be implicitly converted to `nuint` automatically for free.